### DISCUSSION TITLE
Error: Properties layer are unknown

### DISCUSSION THREAD

**Discussion by thany** [vercel/next.js#43556] [Help]
I added some custom babel & webpack configuration, but this error is coming up. I don't understand what's causing this. The rule I'm adding should be correct, as it also exists successfully in another project's webpack config.

Context: we need to integrate some Vue components as part of a gradual migration, so we're thinking of using [vuera](https://github.com/akxcv/vuera) to load Vue components into a Next.js application.

I know it sounds absolutely mental, but migrating the whole application in one go, is much worse in more ways than one 😀

Anyway this is in a vanilla Next.js installation with typescript (just followed the getting started guide). The question is where is this weird error coming from, and how can I fix it please?

`.babelrc`:
```json
{
  "presets": ["next/babel"],
  "plugins": ["vuera/babel"]
}
```

`next.config.js`:
```javascript
const VueLoaderPlugin = require('vue-loader/lib/plugin');

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  webpack: (config, options) => {
    config.module.rules.push({
      test: /\.vue$/,
      loader: 'vue-loader',
    });
    config.plugins.push(new VueLoaderPlugin());
   
    return config;
  },
}

module.exports = nextConfig
```

Error in the console:
```
Error: Compiling RuleSet failed: Properties layer are unknown (at ruleSet[0].rules[0]: [object Object])
    at RuleSetCompiler.error (C:\Projects\Renewi\src\Renewi.Platform.Frontend.Next\node_modules\webpack\lib\rules\RuleSetCompiler.js:373:10)
    at RuleSetCompiler.compileRule (C:\Projects\Renewi\src\Renewi.Platform.Frontend.Next\node_modules\webpack\lib\rules\RuleSetCompiler.js:196:15)
    at C:\Projects\Renewi\src\Renewi.Platform.Frontend.Next\node_modules\webpack\lib\rules\RuleSetCompiler.js:154:9
    at Array.map (<anonymous>)
    at RuleSetCompiler.compileRules (C:\Projects\Renewi\src\Renewi.Platform.Frontend.Next\node_modules\webpack\lib\rules\RuleSetCompiler.js:153:16)
    at RuleSetCompiler.compileRule (C:\Projects\Renewi\src\Renewi.Platform.Frontend.Next\node_modules\webpack\lib\rules\RuleSetCompiler.js:184:30)
    at C:\Projects\Renewi\src\Renewi.Platform.Frontend.Next\node_modules\webpack\lib\rules\RuleSetCompiler.js:154:9
    at Array.map (<anonymous>)
    at RuleSetCompiler.compileRules (C:\Projects\Renewi\src\Renewi.Platform.Frontend.Next\node_modules\webpack\lib\rules\RuleSetCompiler.js:153:16)
    at RuleSetCompiler.compile (C:\Projects\Renewi\src\Renewi.Platform.Frontend.Next\node_modules\webpack\lib\rules\RuleSetCompiler.js:68:22)
```

Versions:

* Node.js 18.12.1
* NPM 8.19.2
* Next.js 13.0.5
* vue-loader 15.9.8
* vuera 0.2.7

**Comment by icyJoseph** [comment]
Could you try with:

```js
const VueLoaderPlugin = require("vue-loader/lib/plugin");

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  webpack(config, options) {
    if (options.nextRuntime !== "edge") {
      config.module.rules.push({
        test: /\.vue$/,
        loader: "vue-loader",
      });
      config.plugins.push(new VueLoaderPlugin());
    }

    return config;
  },
};

module.exports = nextConfig;
```

**Reply by thany** [reply]
I had already tried stuff like that. No matter where I add this rule, it always comes down with this error.

But I still would like to know what is this "layer" it is referring in the error? What is a properties layer??

**Reply by icyJoseph** [reply]
Mmm I think it has to do with Server Components, so it happens in Node.js also then.

Since it fails at rule[0] I guess it might be:

```
{
    "layer": "sc_shared",
    "test": {}
  },
```

Both for browser and nodejs. What Next version are you using? Maybe you should try with `next@12`.
